security/fma-kona-asterisc.md

```diff
@@ -0,0 +1,213 @@
+[cannon]: https://github.com/ethereum-optimism/optimism/tree/develop/cannon
+[op-program]: https://github.com/ethereum-optimism/optimism/tree/develop/op-program
+
+
+## Failure Modes and Recovery Paths
+
+### Kona and Asterisc Hot Path Failure
+
+- **Description:**
+  The `op-program` + `cannon` dispute game breaks and the `kona` + `asterisc` fault dispute game is placed in the hot path.
+  If `kona` + `asterisc` also fail, the permissioned fallback will need to be activated.
+
+- **Risk Assessment:**
+  Low likelihood, high impact.
+  Low likelihood since `op-program` + `cannon` are already running in production and have meaningful improvements on the way that will increase resilience.
+  If `kona` + `asterisc` fails while in the hotpath, the impact is high since the permissioned fallback will need to be triggered.
+  This engages the security council, requires comms, and causes the chain to fallback to a weaker security model.
+
+- **Mitigations:**
+  Offline testing with the VM Runner for both `kona` + `asterisc` as well as `op-program` + `cannon`.
+  The VM Runner also fully tests the integration of the `op-challenger` with these dispute games ([covered in a later section](#Breaking-Backwards-Compatibility-for-the-op-challenger)).
+  Action tests ensuring both proof systems follow spec.
+
+- **Detection:**
+  The game that broke `op-program` + `cannon` would be invalidated since the respected game type would be switched in the dispute game factory.
+  Once `kona` + `asterisc` is the respected game type, the dispute monitor will watch for all new games and will alert if there is an issue with `kona` + `asterisc`.
+  If `kona` + `asterisc` has an issue while in the hot path, and an invalid output / game is played, the dispute monitor and `op-challenger` would alert and detect invalid games.
+  At this point, the permissioned fallback would need to be engaged since both proof systems are in repair.
+
+- **Recovery Path(s):**
+  Likely the only viable recovery path would be the permissioned fallback unless the `op-program` + `cannon` are fixed and able to be placed back into the hotpath.
+
+
+### Divergent Kona Derivation Pipeline
+
+- **Description:**
+  The `kona` derivation pipeline diverged in behavior from the `op-program`'s derivation pipeline.
+
+- **Risk Assessment:**
+  Low likelihood, low impact.
+  Low likelihood as the derivation pipeline is extensively unit tested, run in the VM Runner, and is action tested.
+  Low impact since by default `kona` + `asterisc` are placed in the cold path.
+  [The "Kona and Asterisc Hot Path Failure"](#Kona-and-Asterisc-Hot-Path-Failure) above covers the failure mode for `kona` + `asterisc` when placed in the hot path.
+
+- **Mitigations:**
+  Action tests covering the derivation pipeline are run on every pull request in the [kona] repository.
+  VM Runner executes the derivation pipeline as part of the kona fault proof program running on top of asterisc.
```

Comment:
What does this mean exactly? Is the VM Runner something we're running off chain to verify equivalence against the existing op-program + cannon?

Reply:
VM Runner is a service we execute in our infra that runs the fault proof programs on the fault proof VMs end to end using live chain data, and checks expected results. By running the programs, it adds coverage to the derivation code.